consensus/src/consensus_observer/observer.rs

```diff
@@ -590,6 +590,9 @@ impl ConsensusObserver {
                         error,
                     ))
                 );
+
+                // Send another unsubscription request to the peer
+                self.unsubscribe_from_peer(peer_network_id);
```

Comment:
it looks like we don't return if there's no active_subcription?

Reply:
Yeah, good catch! 😄 I have a fix for this in a future PR, but I'll port it here given that we're already touching this code.

Reply:
Added a modified fix below. Will refactor some of this code in the future, though. 😄